scripts/pc2_interactive.sh

```diff
@@ -244,13 +246,16 @@ do
 			if test -d /proc/$contestantpid
 			then
 				REPORT Contestant PID $submissionpid has not finished - killing it
-				kill -9 "$submissionpid"
 			fi
 			# This is just determines if the program ran, not if it's correct.
 			# The result file has the correctness in it.
 			# We only do this if the contestant program has not finished yet.
 			COMMAND_EXIT_CODE=0
 		fi
+
+		# Kill off submission pgrp
+		kill -9 -"$submissionpid"
```

Comment:
again the - here on the pid?  are you expecting the submissionpid to 1?

Reply:
No.  It' is the process group of the submission.